projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,295 @@
+          public void onClick(View view) {
+            getWriteStoragePermission();
+          }
+        });
+  }
+
+  private void createPdf() {
+    // TODO: Success reading license file only from "res/raw". Perhaps should be from a different
+    // dedicated folder.
+    // Activate iText license
+    LicenseKey.loadLicenseFile(getResources().openRawResource(R.raw.itextkey));
+    try {
+      HtmlConverter.convertToPdf(
+          fillTemplate(),
+          new FileOutputStream(Paths.get(lawsuitOutputPath, getPdfFilename()).toString()));
+      Toast.makeText(LawsuitPdfActivity.this, "Lawsuit created!", Toast.LENGTH_LONG).show();
+    } catch (IOException e) {
+      Log.w(TAG, "Error creating PDF Lawsuit");
+      e.printStackTrace();
+    }
+  }
+
+  private String fillTemplate() {
+
+    String lawsuit = readTemplate();
+
+    // General form fields
+    lawsuit =
+        lawsuit.replace("&lt;claimCase&gt;", sentHaser ? claimCaseHaser : claimCaseSubscription);
+    lawsuit = lawsuit.replace("&lt;claimAmount&gt;", claimAmount);
+    lawsuit = lawsuit.replace("&lt;spamType&gt;", spamType);
+    lawsuit =
+        lawsuit.replace(
+            "&lt;moreThanFiveLawsuits&gt;",
+            sentMoreThanFiveLawsuits ? moreThanFiveLawsuits : lessThanFiveLawsuits);
+    lawsuit = lawsuit.replace("&lt;undefined&gt;", "");
+    lawsuit = lawsuit.replace("&lt;receivedSpamDate&gt;", DATE_FORMATTER.format(receivedSpamDate));
+    lawsuit = lawsuit.replace("&lt;currentDate&gt;", DATE_FORMATTER.format(new Date()));
+
+    // User fields
+    lawsuit = lawsuit.replace("&lt;userFullName&gt;", userPrivateName + " " + userLastName);
+    lawsuit = lawsuit.replace("&lt;userId&gt;", userID);
+    lawsuit = lawsuit.replace("&lt;userAddress&gt;", userAddress);
+    lawsuit = lawsuit.replace("&lt;userPhone&gt;", userPhone);
+    lawsuit = lawsuit.replace("&lt;userFax&gt;", userFax);
+
+    // First spam company
+    lawsuit = lawsuit.replace("&lt;companyName&gt;", companyName);
+    lawsuit = lawsuit.replace("&lt;companyId&gt;", companyId);
+    lawsuit = lawsuit.replace("&lt;companyAddress&gt;", companyAddress);
+    lawsuit = lawsuit.replace("&lt;companyPhone&gt;", companyPhone);
+    lawsuit = lawsuit.replace("&lt;companyFax&gt;", companyFax);
+
+    // Second spam company
+    lawsuit = lawsuit.replace("&lt;company2Name&gt;", company2Name);
+    lawsuit = lawsuit.replace("&lt;company2Id&gt;", company2Id);
+    lawsuit = lawsuit.replace("&lt;company2Address&gt;", company2Address);
+    lawsuit = lawsuit.replace("&lt;company2Phone&gt;", company2Phone);
+    lawsuit = lawsuit.replace("&lt;company2Fax&gt;", company2Fax);
+
+    return lawsuit;
+  }
+
+  private String readTemplate() {
+
+    try {
+      return new String(
+          Files.readAllBytes(Paths.get(lawsuitTemplateFilePath)), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      Log.w(TAG, "Error reading lawsuit's template.");
+      e.printStackTrace();
+      return null;
```

Comment:
If this returns null, you're going to get an NPE on `fillTemplate()`.
I think you should have a try-catch in `fillTemplate()` to handle that.
But, then this method just becomes `Files.readAllBytes(Paths.get(lawsuitTemplateFilePath)), StandardCharsets.UTF_8);`, which in that case, I suggest merging it with `fillTemplate()` :)

Reply:
Fixed :)